src/frontends/paddle/src/op/gather_nd.cpp

```diff
@@ -0,0 +1,22 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "default_opset.hpp"
+#include "openvino/frontend/paddle/node_context.hpp"
+
+namespace ov {
+namespace frontend {
+namespace paddle {
+namespace op {
+NamedOutputs gather_nd(const NodeContext& node) {
+    const auto data_node = node.get_input("X");
+    const auto index_node = node.get_input("Index");
+    PADDLE_OP_CHECK(node, index_node.get_partial_shape().size() > 0, "Index shape must be non-empty");
```

Comment:
```suggestion
    auto shape = index_node.get_partial_shape();
    for (const auto& dim : shape) {
        if (dim.is_static() && dim.get_length() == 0)
            PADDLE_OP_CHECK(node, false, "zero dimension is not allowed for gather_nd Index");
    }
```
suggest use this way to meet both dynamic and static  shape conditions